### DISCUSSION TITLE
Eslint plugin template custom rule get info from component ts file

### DISCUSSION THREAD

**Discussion by Harpush** [angular-eslint/angular-eslint#1295] [Q&A] [answered]
I want to create a template rule which needs ts file info. I wish to error on every observable with async pipe coming from a store service that isn't used inside viewModel. For that to work I need to access the TS source file from the template and I can't find any reference in the template AST. Is it possible? Or I need to create the TS source file myself from file name manipulation and try and match the variable names? 
I know eslint isn't really built for multiple files linting but I can see no other way.

**Comment by JamesHenry** [comment, marked as answer]
It's not technically possible right now, we only have syntactic data about the template AST in isolation. You might wish to look into Angular's native concept of diagnostics: https://angular.dev/extended-diagnostics#new-diagnostics
